logic/src/commonMain/kotlin/com/wire/kalium/logic/data/publicuser/UserSearchApiWrapper.kt

```diff
@@ -0,0 +1,72 @@
+package com.wire.kalium.logic.data.publicuser
+
+import com.wire.kalium.logic.NetworkFailure
+import com.wire.kalium.logic.data.id.IdMapper
+import com.wire.kalium.logic.di.MapperProvider
+import com.wire.kalium.logic.functional.Either
+import com.wire.kalium.logic.functional.map
+import com.wire.kalium.logic.wrapApiRequest
+import com.wire.kalium.network.api.contact.search.UserSearchApi
+import com.wire.kalium.network.api.contact.search.UserSearchRequest
+import com.wire.kalium.network.api.contact.search.UserSearchResponse
+import com.wire.kalium.persistence.dao.ConversationDAO
+import kotlinx.coroutines.flow.firstOrNull
+
+interface UserSearchApiWrapper {
```

Comment:
Let's try to make these `internal`, so they don't leak out to consumers of Kalium.

```suggestion
internal interface UserSearchApiWrapper {
```